petastorm/tests/test_end_to_end.py

```diff
@@ -768,3 +768,20 @@ def read_one_row():
             results = [f.result() for f in futures]
             assert len(results) == len(synthetic_dataset.data)
             assert set(r.id for r in results) == set(d['id'] for d in synthetic_dataset.data)
+
+
+def test_should_fail_if_reading_out_of_context_manager(synthetic_dataset):
+    with make_reader(synthetic_dataset.url, workers_count=1) as reader:
+        next(reader)
+
+    with pytest.raises(RuntimeError, match='Trying to read a sample.*'):
+        next(reader)
```

Comment:
Shouldn't this error be that the `reader` variable is undefined since `reader` went out of scope?

Reply:
Apparently, the variable is still defined after the context block. 
```
In [1]: with open('/tmp/zzz', 'wt') as f:
   ...:     pass
   ...: 

In [2]: f
```